encodercontrol.cpp

```diff
@@ -41,6 +45,9 @@ EncoderControl::EncoderControl(QObject *parent ) : QObject( parent )
 #endif
 }
 
+// QA: why you don't use reference as argument, kind of QString& str
+// QA: as I understand, each time you call this method, you copy the input arg 
+// QA: in your case, as you don't change str, you may use const QString& str 
```

Comment:
You're quite right. I copy this string because in port.cpp I use asynchronous mode. This is how I care about nice data.